pkg/minikube/assets/addons.go

```diff
@@ -196,22 +196,53 @@ var Addons = map[string]*Addon{
 	}, false, "registry-creds"),
 }
 
-func AddMinikubeDirToAssets(minipath string, vmpath string, assetList *[]CopyableFile) {
-	// loop over $MINIKUBE_HOME/minipath and add them to assets
-	searchDir := constants.MakeMiniPath(minipath)
-	err := filepath.Walk(searchDir, func(miniFile string, f os.FileInfo, err error) error {
-		isDir, err := util.IsDirectory(miniFile)
-		if err == nil && !isDir {
-			f, err := NewFileAsset(miniFile, vmpath, filepath.Base(miniFile), "0640")
-			if err == nil {
-				*assetList = append(*assetList, f)
+func AddMinikubeDirAssets(assets *[]CopyableFile) error {
+	if err := addMinikubeDirToAssets(constants.MakeMiniPath("addons"), constants.AddonsPath, assets); err != nil {
+		return errors.Wrap(err, "adding addons folder to assets")
+	}
+	if err := addMinikubeDirToAssets(constants.MakeMiniPath("files"), "", assets); err != nil {
+		return errors.Wrap(err, "adding files rootfs to assets")
+	}
+
+	return nil
+}
+
+// AddMinikubeDirToAssets adds all the files in the basedir argument to the list
+// of files to be copied to the vm.  If vmpath is left blank, the files will be
+// transferred to the location according to their relative minikube folder path.
+func addMinikubeDirToAssets(basedir, vmpath string, assets *[]CopyableFile) error {
+	err := filepath.Walk(basedir, func(hostpath string, info os.FileInfo, err error) error {
+		isDir, err := util.IsDirectory(hostpath)
+		if err != nil {
+			return errors.Wrapf(err, "checking if %s is directory", hostpath)
+		}
+		if !isDir {
+			if vmpath == "" {
+				rPath, err := filepath.Rel(basedir, hostpath)
+				if err != nil {
+					return errors.Wrap(err, "generating relative path")
+				}
+				rPath = filepath.Dir(rPath)
```

Comment:
Not sure this will work with Windows as you stated.  Should be able to massage the paths though.

Reply:
I don't think the tests will cover this, I don't think any integration tests use the .minikube/files dir.